### ISSUE TITLE
K4-System 2.0.1 - CSSharp: Native ab9aa921 was invoked on a non-main thread

### ISSUE THREAD

**Issue by mrdiega** [K4ryuu/K4-System#67]
K4-System 2.0.1
css v110
metamod 2.0 - build 1271

spamming errors after round end
`CSSharp: Native ab9aa921 was invoked on a non-main thread`
![image](https://github.com/K4ryuu/K4-System/assets/79888956/2f166498-a44f-423b-bde5-4773520a5cdd)

**Comment by K4ryuu** [comment]
Greetings,
I am going to check the problem after work. 
Thank you

**Comment by mrdiega** [comment]
> Greetings, I am going to check the problem after work. Thank you
@K4ryuu 

![image](https://github.com/K4ryuu/K4-System/assets/79888956/99e29008-f208-4b2d-a46d-5a730a8ad39c)
The problem still exists in 2.0.2

upd.
I played with a player for several minutes 1 on 1 and then the server crashed with the error:
```
Process terminated. A callback was made on a garbage collected delegate of type 'System.Private.CoreLib!System.Action::Invoke'.

[screen is terminating]
```
In addition, sometimes more points are awarded and taken away than expected

**Comment by K4ryuu** [comment]
The bug has been solved within the next update (V2.0.3). We are testing it further and will be released tomorrow with lot of patches and bug fixes.
